core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -476,7 +493,29 @@ impl<Block: BlockT> Inner<Block> {
 		}
 	}
 
-	fn consider_vote(&self, round: Round, set_id: SetId) -> Consider {
+	fn consider_vote(&mut self, round: Round, set_id: SetId, msg: &SignedMessage<Block>, id: &PeerId) -> Consider {
+		let mut tally = self.votes_tally.entry(id.clone()).or_insert(Default::default());
```

Comment:
logic here is wrong -- this is tallying equivocations per _peer_ but note that peers in the gossip network are not the same as _voters_ who issue votes. Even if there are 10 voters, we could have 1000 peers.

Reply:
The other broken piece of logic here is that the vote tallies aren't scoped by round.

Reply:
> this is tallying equivocations per peer but note that peers in the gossip network are not the same as voters who issue votes

It appeared to me that the code in `communication` was not aware of voters, just of peers, but I'll take another look and see if I can use voters here instead of peers. 

> the vote tallies aren't scoped by round

The tallies are reset there when a new round is started: https://github.com/paritytech/substrate/pull/2479/files#diff-1c4fd5ec8a616ff527309711747b39d3R468

Is that inacurate?